src/wmtk/Mesh.hpp

```diff
@@ -338,33 +338,166 @@ class Mesh : public std::enable_shared_from_this<Mesh>
     //============================
     // MultiMesh interface
     //============================
+    //
+    /**
+     * @brief return if this mesh is the root of a multimesh tree
+     */
     bool is_multi_mesh_root() const;
+    /**
+     * @brief returns a reference to the root of a multimesh tree
+     */
     Mesh& get_multi_mesh_root();
-    std::vector<std::shared_ptr<Mesh>> get_child_meshes() const;
+    /**
+     * @brief returns a const reference to the root of a multimesh tree
+     */
     const Mesh& get_multi_mesh_root() const;
+
+    /**
+     * @brief returns the direct multimesh child meshes for the current mesh
+     */
+    std::vector<std::shared_ptr<Mesh>> get_child_meshes() const;
+
+    /**
+     * @brief returns a unique identifier for this mesh within a single multimesh structure
+     */
     std::vector<long> absolute_multi_mesh_id() const;
+
+
+    /**
+     * @brief register a mesh as the child of this mesh
+     *
+     *  The parameter mesh_tuples is a sequence of {a,b} where a is a tuple of
+     *  this mesh and b is a tuple of the child mesh.
+     *  The tuple b is assumed to encode a top dimension simplex in the child
+     *  mesh, and a is any tuple that corresponds to that simplex in the parent
+     *  mesh
+     *
+     * @param child_mesh the mesh that will become a child of this mesh
+     * @param mesh_tuples a sequence of corresponding tuples between meshes
+     */
     void register_child_mesh(
         const std::shared_ptr<Mesh>& child_mesh,
         const std::vector<std::array<Tuple, 2>>& map_tuples);
 
-    // a generic map interface between pairs of mesh in a single multi-mesh structure
+    /**
+     * @brief maps a simplex from this mesh to any other mesh
+     *
+     *
+     * Generic interface for mapping between two arbitrary meshes in a multi-mesh structure
+     * Note that this finds ALL versions of a simplex, potentially crossing over topological
+     * features above the pairs of simplices being mapped. For instance, if we map a trimesh seam
+     * edge to itself using this interface it will find the edge on the other side of the seam. If
+     * more granular mappings are required consider manually navigating the tree with map_to_parent
+     * and map_to_child, which of course require a more particular understanding on how a simplex is
+     * mapped. throws if two meshes are not part of the same multi-mesh structure
+     *
+     *
+     * @param the mesh a simplex should be mapped to
+     * @param the simplex being mapped to the child mesh
```

Comment:
```
@param other_mesh the mesh...
@param my_simplex the simplex...
```
You forgot to write the parameter name also in other occasions but iirc it is necessary for a proper Doxygen documentation.

Reply:
i did this on a few - but perhaps we should enforce doxygen warnings too :D